extensions/metrics-cluster-feature/src/metrics-settings.tsx

```diff
@@ -0,0 +1,276 @@
+/**
+ * Copyright (c) 2021 OpenLens Authors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+import React from "react";
+import { Component, Catalog, K8sApi } from "@k8slens/extensions";
+import { observer } from "mobx-react";
+import { computed, observable } from "mobx";
+import { MetricsFeature, MetricsConfiguration } from "./metrics-feature";
+
+interface Props {
+  cluster: Catalog.KubernetesCluster;
+}
+
+@observer
+export class MetricsSettings extends React.Component<Props> {
+  @observable featureStates = {
+    prometheus: false,
+    kubeStateMetrics: false,
+    nodeExporter: false
+  };
+  @observable canUpgrade = false;
+  @observable upgrading = false;
+  @observable changed = false;
+  @observable inProgress = false;
+
+  config: MetricsConfiguration = {
+    prometheus: {
+      enabled: false
+    },
+    persistence: {
+      enabled: false,
+      storageClass: null,
+      size: "20G",
+    },
+    nodeExporter: {
+      enabled: false,
+    },
+    retention: {
+      time: "2d",
+      size: "5GB",
+    },
+    kubeStateMetrics: {
+      enabled: false,
+    },
+    alertManagers: null,
+    replicas: 1,
+    storageClass: null,
+  };
+  feature: MetricsFeature;
+
+  @computed get isTogglable() {
+    if (this.inProgress) return false;
+    if (!this.props.cluster.status.active) return false;
+    if (this.canUpgrade) return false;
+    if (!this.isActiveMetricsProvider) return false;
+
+    return true;
+  }
+
+  get metricsProvider() {
+    return this.props.cluster.spec?.metrics?.prometheus?.type || "";
+  }
+
+  get isActiveMetricsProvider() {
+    return (!this.metricsProvider || this.metricsProvider === "lens");
+  }
+
+  async componentDidMount() {
+    this.feature = new MetricsFeature(this.props.cluster);
+
+    await this.updateFeatureStates();
+  }
```

Comment:
This `await` doesn't really do anything because `componentDidMount` is treated as if it returns `void`.